### ISSUE TITLE
Single stream rgb is working but double stream not 

### ISSUE THREAD

**Issue by Scienceseb** [pmorerio/admd#4]
So the loading of the rgb/resnet50 is a success but when Loading pretrained depth/resnet50 I got the following error: **The passed save_path is not a valid checkpoint: model/depth.** 

_I'm running exactly the same code as you no modification except the path for the data and models._ 

But in the folder model/resnet50 I got:
resnet_v1_50.ckpt,
resnet_v1_50_rgb.ckpt.data-00000-of-00001,
resnet_v1_50_rgb.ckpt.index,
resnet_v1_50_rgb.ckpt.meta
resnet_v1_50_depth.ckpt.data-00000-of-00001,
resnet_v1_50_depth.ckpt.index,
resnet_v1_50_depth.ckpt.meta

 Interestingly the single stream with the depth is working perfectly fine.

**Comment by pmorerio** [comment]
So, not really sure what  the problem is.
Single rgb works fine
Single depth works fine
Double stream does not work, is that correct?

**Comment by pmorerio** [comment]
After you train `rgb` and `depth`, you should have the trained checkpoints saved in the `model` folder. 
When you train the `double_stream`, the two branches of the network are initialized with those checkpoints. From your error, it look like you do not have the depth checkpoint there.

![image](https://user-images.githubusercontent.com/6783835/66639839-f62c1d80-ec17-11e9-9fb3-f03e56b4f572.png)

**Comment by Scienceseb** [comment]
I think my path was wrong. Also when in training it's written test acc. **do you really check the accuracy on the test set at every epoch?**

**Comment by pmorerio** [comment]
Well at the beginning I was just  testing at the end with the `--mode test_xxx` options. Early stopping is done when loss is nearly stable since there is no validation set for this dataset. I inserted the test check during training just to verify everything was ok. **But in no way test accuracy is used for validation**.

**Comment by Scienceseb** [comment]
Ok it's working but I dont really undertstand I dont get performance like you there must be a bug, just to recap what I do for training the double stream, I first train RGB and depth single stream after I start the training for the double stream, at start the test. acc is 54% but after first epoch it decrease to 37% and at the end of the 2000 iterations its even lower, around 23% acc. I didnt change anything in your code. 

![Screenshot from 2019-10-12 10-25-29](https://user-images.githubusercontent.com/26779418/66702910-a5c9c400-ecda-11e9-9beb-b11eaf239650.png)
![Screenshot from 2019-10-12 10-35-35](https://user-images.githubusercontent.com/26779418/66703029-0a395300-ecdc-11e9-9b9e-d453d4feee70.png)

**Comment by Scienceseb** [comment]
![Screenshot from 2019-10-12 10-25-29](https://user-images.githubusercontent.com/26779418/66702910-a5c9c400-ecda-11e9-9beb-b11eaf239650.png)

**Comment by Scienceseb** [comment]
![Screenshot from 2019-10-12 10-35-35](https://user-images.githubusercontent.com/26779418/66703029-0a395300-ecdc-11e9-9b9e-d453d4feee70.png)

**Comment by pmorerio** [comment]
Hi, as we state in the paper, finetuning the two-stream does not help much and is often harmful. 
Simply taking the two separate streams (trained independently) and average logits works.
For NYUD I think I only finetuned for 1000 iterations with a very low learning rate.
Please see https://github.com/pmorerio/admd/blob/754d5cc8d3aeae252a5c2cb16f66d5aac2d31de6/NYUD/run.sh#L11
Running this will finetune for a tiny bit and save the two stream in a single model for testing and further use.

Do you get a reasonable accuracy for the two separate RGB  and Depth stream (cf. Table 5 in the paper)?

**Comment by Scienceseb** [comment]
I'm gonna give you numbers tomorrow or after tomorrow, but it was less than what you put in your paper to be frank.

**Comment by pmorerio** [comment]
This is what I get, which is the number in the paper.

![image](https://user-images.githubusercontent.com/6783835/66809290-59c28d80-ef2d-11e9-9cd7-bd05fd3af60c.png)

**Comment by Scienceseb** [comment]
What is your training strategy ? What do I have to train: single stream RGB, single stream Depth, double stream after that (?) and finally Hallucination network ? I followed your paper, but certainly I was wrong somewhere.

**Comment by pmorerio** [comment]
Yes, that is the idea. In the file `run.sh` you have everything, also some other baselines such as modDrop and rgb ensemble.

**Comment by pmorerio** [comment]
Closing this since inactive for a week
